### ISSUE TITLE
I've been working on another parser for groovy

### ISSUE THREAD

**Issue by murtaza64** [Decodetalkers/tree-sitter-groovyscript#5]
Hi, I just noticed that this parser already existed after spending like two weeks working on my own version from scratch. Would you like to check it out at https://github.com/murtaza64/tree-sitter-jenkins?

(It's called Jenkins but there's very few jenkins-specific features, so it can be adapted to be a general groovy parser)

I haven't tested your implementation but mine seems to, at face value, be more feature complete as of now. 

Would you be interested in merging our projects so that the upstream parser in nvim-treesitter is more feature complete?

**Comment by Decodetalkers** [comment]
You can make pr now. I write this parse just for android development .And I also handle decorate, generics_class and etc, it works good, if just come from gradle and general groovy script, it has no error when parse these, I am satisfied with it. But I do not know groovy very much. so I just make it performs well on android gradle script. so , yes , if you think this is not good enough, make pr now, and I also willing to make you the maintainer if you willing to

**Comment by Decodetalkers** [comment]
Can you let me take a look at the highlight of yours? I will take a try this weekend.. and of course, you can make a pr, because here , there are some important ci. have you try that in gradle script?

**Comment by Decodetalkers** [comment]
Well, yours are good enough, please make pr now, and I will push scheme update to nvim-treesitter

**Comment by Decodetalkers** [comment]
And 
![image](https://github.com/Decodetalkers/tree-sitter-groovy/assets/60290287/d8c487a0-93b9-4208-9368-425edeb6de7b)

can this be highlighted?

**Comment by Decodetalkers** [comment]
https://github.com/apache/groovy/blob/3b1b1cb3f2f78432a86ed7a7776d3cc13e107366/subprojects/groovy-groovysh/src/main/groovy/org/apache/groovy/groovysh/completion/antlr4/InfixKeywordSyntaxCompleter.groovy#L4

this is not parse well enought

**Comment by Decodetalkers** [comment]
and also Generics

**Comment by Decodetalkers** [comment]
My parse have already handle most error, and decorate and generic.. can you help add support for them?

**Comment by Decodetalkers** [comment]
Emm, there is too many error.. but it is still good. though to many error

**Comment by murtaza64** [comment]
Yes, I will work on generics, decorators and stuff before making any PRs. Just wanted to gauge your interest in merging projects. When it's ready, would you prefer to push my version to upstream?

> can this be highlighted?

You mean `dependencies`? Sure. But I am not sure if the current way of highlighting jenkins-specific symbols is great because they will still be highlighted even when not using jenkins, and the same problem will happen for gradle. Any ideas here?

Thanks for sharing examples of files where it doesn't work well.

Also, feel free to contribute to the project. Or I might take a look at your error/decorator/generics parsing and bring it into mine.

**Comment by murtaza64** [comment]
Today I'll look at your code and see how we can merge the parts you've implemented with the parts I have.

**Comment by fussybeaver** [comment]
I'm quite excited to see this and hope you merge your work streams...

**Comment by murtaza64** [comment]
I've implemented annotations and generics, please take another look! (no highlighting yet) (Haven't got around to `import` and `package` yet)

**Comment by Decodetalkers** [comment]
Cool grammar, can you make pr now? Just replace the gammer, test, and c code under src

**Comment by Decodetalkers** [comment]
And after this pr, you should also add your name to package.json..and you should make a pr to nvim-treesitter

**Comment by Decodetalkers** [comment]
And do you write any groovy script? I think we need someone who are using this script to maintain it.. before I keep this gammer simple enough to make it works under most case

**Comment by Decodetalkers** [comment]
You also need to test it yourself. Replace the name in package.json and gammer.js to groovy, and test it. You also need find some groovy examples

**Comment by Decodetalkers** [comment]
Emm, have you test grammer like
```groovy
foreach { map -> 
  print(map)
}
```

**Comment by Decodetalkers** [comment]
Ping..

**Comment by murtaza64** [comment]
Sorry, I've been busy with work

- Yes, I work with groovy daily (in Jenkins)
- I have to implement closures still
- I think it's better to switch the upstream config to use my repo instead of completely overwriting this repo, thoughts?
- I will add more full-script tests and stuff once this is production ready. I've been using it on my machine for a few days and most things work, but I still keep finding bugs here and there so I'm not ready to merge it. If you would like to help with development, feel free to make PRs in my repo

**Comment by murtaza64** [comment]
Sorry it's still been very busy at work. I will try to finish this up over the holidays!

**Comment by murtaza64** [comment]
@Decodetalkers I think my parser is mature enough to replace this one now. Would you be OK with switching the upstream parser to mine instead of yours?

**Comment by Decodetalkers** [comment]
Ok, please
